### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by BadgerCode**
- Rabbit config
- Connecting to rabbit
- Creating a basic header exchange queue
- A blocking consumer
- A publisher

### CODE REVIEW

**Comment by BadgerCode** [comment]
Look at SQL connection pooling.

Look into Connection failing but things on connection not being notified
https://www.rabbitmq.com/reliability.html

Fix the problem with rabbit messages of the wrong format deserialising fine, but nulling out all the properties.

**Comment by BadgerCode** [comment]
Try out writing a project with this with unit tests.
Stop worrying about making the interfaces not Rabbit-specific.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,235 @@
...
+
+var messageHeaders = new Dictionary<string, string>
+{
+    { "Command", "UpdateUser" },
+    { "Version", "1.0" }
+};
+var messageBody = new UpdateUserCommand
+{
+    UserId = Guid.NewGuid(),
+    FirstName = "John",
```
Comment (BnSmth) [inline README.md:111]: Fail.

Comment (BadgerCode) [inline README.md:111]: You so vain.

**Pull request merged**